code/test-utils/src/test/java/com/adobe/marketing/tester/util/JSONAssertsParameterizedTests.kt

```diff
@@ -0,0 +1,416 @@
+    JSONAssertsParameterizedTests.ExpectedArrayLargerTest::class,
+    JSONAssertsParameterizedTests.ExpectedDictionaryLargerTest::class
+)
+class JSONAssertsParameterizedTests {
+    @RunWith(Parameterized::class)
+    class ValueMatchingTest(private val expected: Any, private val actual: Any) {
+        companion object {
+            @JvmStatic
+            @Parameterized.Parameters(name = "{index}: test with expected={0}, actual={1}")
+            fun data(): Collection<Array<Any>> {
+                return listOf(
+                    Array(2) { 5 }, // Int
+                    Array(2) { 5.0 }, // Double
+                    Array(2) { true }, // Boolean
+                    Array(2) { "a" }, // String
+                    Array(2) { "안녕하세요" }, // Non-Latin String
+                    Array(2) { JSONArray() },
+                    Array(2) { JSONArray("""[[[]]]""") }, // Nested arrays
+                    Array(2) { JSONObject() },
+                    Array(2) { JSONObject.NULL },
+                    Array(2) { JSONObject("""{ "key1": 1 }""") }, // Key value pair
+                    Array(2) { JSONObject("""{ "key1": { "key2": {} } }""") }, // Nested objects
+                )
+            }
+        }
+
+        @Test
+        fun `should match basic values`() {
+            assertEqual(expected, actual)
+            assertExactMatch(expected, actual)
+            assertTypeMatch(expected, actual)
+        }
+    }
+
+    @RunWith(Parameterized::class)
+    class TypeMatchingTest(private val expected: Any, private val actual: Any) {
+        companion object {
+            @JvmStatic
+            @Parameterized.Parameters(name = "{index}: test with expected={0}, actual={1}")
+            fun data(): Collection<Array<Any>> {
+                return listOf(
+                    arrayOf(5, 10), // Int
+                    arrayOf(5.0, 10.0), // Double
+                    arrayOf(true, false), // Boolean
+                    arrayOf("a", "b"), // String
+                    arrayOf("안", "안녕하세요"), // Non-Latin String
+                    arrayOf(JSONObject("""{ "key1": 1 }"""),
+                            JSONObject("""{ "key1": 2 }""")), // Key value pair
+                    arrayOf(JSONObject("""{ "key1": { "key2": "a" } }"""),
+                            JSONObject("""{ "key1": { "key2": "b", "key3": 3 } }""")), // Nested partial by type
+                )
+            }
+        }
+
+        @Test
+        fun `should match only by type for values of the same type`() {
+            Assert.assertThrows(AssertionError::class.java) {
+                assertEqual(expected, actual)
+            }
+            Assert.assertThrows(AssertionError::class.java) {
+                assertExactMatch(expected, actual)
+            }
+            assertTypeMatch(expected, actual)
+        }
+    }
+
+    @RunWith(Parameterized::class)
+    class FlexibleTypeMatchingTest(private val expected: Any, private val actual: Any) {
```

Comment:
This one seems to test agains empty JSONArray/JSONObject, does this need to be renamed or updated?